src/t_string.c

```diff
@@ -102,6 +104,31 @@ void setGenericCommand(client *c,
 
     found = (lookupKeyWrite(c->db, key) != NULL);
 
+    /* Handle the IFEQ conditional check */
+    if (flags & OBJ_SET_IFEQ && found) {
+        robj *current_value = lookupKeyRead(c->db, key);
+
+        if (current_value == NULL || current_value->type != OBJ_STRING || checkType(c, comparison, OBJ_STRING) != 0) {
+            if (!(flags & OBJ_SET_GET)) {
+                addReplyError(c, "value(s) must be present or string");
+            }
+            return;
+        }
```

Comment:
In this block `found` is true, so I believe we don't need to check `current_value` as NULL. We don't support key with `NULL` value in Valkey.
Also, `comparison` object is from the client input, so I think we can be assured it would be of string type. We should use the checkType method call for `current_value`. We can simplify it to:

```suggestion
        if (checkType(c, current_value, OBJ_STRING)) {
            return;
        }
```

Reply:
Used the type to verify the current_value's data type